File: python/exercise3.py
def count_sheeps(sheep):
    return len([item for item in sheep if item])  

def test_count_sheeps():
    arrayTest = [False, True, True, True, None, None]
    if count_sheeps(arrayTest) == 3:
        print("OK")
    else:
        print("NOK")

test_count_sheeps()

File: java/Exercise3.java
import java.util.*;

public class Exercise3 {
    public int countSheeps(Boolean[] arrayOfSheeps) {
        int totalSheeps = 0;
        for (Boolean sheep: arrayOfSheeps) {
            if (sheep == null) {
                continue;
            } else if (sheep) {
                totalSheeps++;
            }
        }
        return totalSheeps;
    }
    public static void main(String[] args) {
        Boolean[] array1 = {
            true,  true,  true,  false,
            true,  true,  true,  true ,
            true,  false, true,  false,
            true,  false, false, true ,
            true,  true,  true,  true ,
            false, false, true,  true 
        };
        System.out.println(new Exercise3().countSheeps(array1));
    }
}